grabbit/src/main/groovy/com/twcable/grabbit/client/servlets/GrabbitJobServlet.groovy

```diff
@@ -107,37 +90,42 @@ class GrabbitServlet extends SlingAllMethodsServlet {
         final clientUsername = request.remoteUser
         final GrabbitConfiguration configuration
         try {
-            configuration = GrabbitConfiguration.create(input)
+            configuration = GrabbitConfiguration.create(configurationInput)
         } catch(ConfigurationException ex) {
             log.warn "Bad configuration for request. ${ex.errors.values().join(',')}"
-            response.status = HttpServletResponse.SC_BAD_REQUEST
+            response.status = SC_BAD_REQUEST
             response.contentType = "application/json"
-            response.writer.write(new JsonBuilder(ex.errors).toString())
+            response.writer.write("Bad configuration for request. More detail: \n\n ${new JsonBuilder(ex.errors).toString()}")
             return
         }
         Collection<Long> jobExecutionIds = clientService.initiateGrab(configuration, clientUsername)
         log.info "Jobs started : ${jobExecutionIds}"
-        response.status = HttpServletResponse.SC_OK
+        log.info "Transaction started ${configuration.transactionID}"
+        response.status = SC_OK
         response.contentType = "application/json"
+        response.addHeader("GRABBIT_TRANSACTION_ID", "${configuration.transactionID}")
         response.writer.write(new JsonBuilder(jobExecutionIds).toString())
     }
 
 
-    private String getJsonString(String jobId) {
+    private String getJsonResponseForJobStatus(String jobId) {
         final JobExplorer jobExplorer = configurableApplicationContext.getBean("clientJobExplorer", JobExplorer)
         if (jobId.isNumber()) {
             //Returns Status for A Job
-            final ClientJobStatus status = ClientJobStatus.get(jobExplorer, Long.valueOf(jobId))
-            new JsonBuilder(status).toString()
+            try {
+                final ClientJobStatus status = ClientJobStatus.get(jobExplorer, Long.valueOf(jobId))
+                return new JsonBuilder(status).toString()
+            }
+            catch(NonExistentJobException ex) {
+                log.warn "Queried for job with ID ${jobId} however none was found!"
```

Comment:
Calm down. I'm sure the caller will fix their mistake. No need to get so worked up...